src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -1264,8 +1264,9 @@ static void lb_call_on_retry_timer_locked(grpc_exec_ctx* exec_ctx, void* arg,
       gpr_log(GPR_INFO, "Restaring call to LB server (grpclb %p)",
               (void*)glb_policy);
     }
-    GPR_ASSERT(glb_policy->lb_call == NULL);
-    query_for_backends_locked(exec_ctx, glb_policy);
+    if (glb_policy->lb_call == NULL) {
```

Comment:
I assume that the problem here is the race condition I spoke about in my talk at the team meeting last week?  In other words, this happens when the timer closure is scheduled before we cancel the timer but not actually run until after we recreate the call?

If that's not the case, please explain what actually triggers this condition.  Thanks!

Reply:
Precisely!